test/common/http/path_utility_fuzz_test.cc

```diff
@@ -0,0 +1,37 @@
+#include "common/http/path_utility.h"
+
+#include "test/common/http/path_utility_fuzz.pb.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+
+namespace Envoy {
+namespace Fuzz {
+namespace {
+DEFINE_PROTO_FUZZER(const test::common::http::PathUtilityTestCase& input) {
+  switch (input.path_utility_selector_case()) {
+  case test::common::http::PathUtilityTestCase::kCanonicalPath: {
+    auto request_headers = fromHeaders<Http::TestRequestHeaderMapImpl>(
+        input.canonical_path().request_headers(), {},
+        {":path"}); // needs to have path header in order to be valid
+    Http::PathUtil::canonicalPath(request_headers);
```

Comment:
is there any other validation that we can do on this? possible just verifying that the header map contains a non-empty path header string after applying this function?

Reply:
This fails quite a bit, simply due to the random nature of the fuzzing engine. Should I do a release assert here? I had a simple EXPECT_TRUE from the returned boolean, and also an EXPECT_NE to an empty string.

Reply:
Validating the return value will fail, like you said, but validating that the function preserves a path header like you have will always work